### DISCUSSION TITLE
Any examples on iterating through only normal text nodes?

### DISCUSSION THREAD

**Discussion by sheldonhull** [yuin/goldmark#259] [Q&A] [answered]
I am trying to do some text parsing of _only_ the regular text nodes.

For example, I'm doing string replacement to break sentences into new lines, but I don't want to rely on regex parsing in Go for detecting if the text block is part of a table, or quote, or other things that would break if I did just a standard line break.

I see lots of documentation on Tables, Headers, and other specific structures, but is there any example that would show how to take a markdown file and only perform the action if it's a normal text node?

Current References I'm finding:

- https://github.com/yuin/goldmark/issues/252#issue-1020905615
- https://github.com/yuin/goldmark/discussions/215 -> Useful example of ast text node addition.
- [Text Block](https://pkg.go.dev/github.com/yuin/goldmark@v1.4.1/ast#TextBlock) but couldn't find anything showing that NodeKind referenced a way to iterate through something like `for _, item := range Nodes { if item.NodeKind == nodeKind.TextBlock}` etc. Hoping someone has a simple example of this I can look at.

**Comment by mmiranda** [comment, marked as answer]
I think you can follow my example here https://github.com/yuin/goldmark/discussions/254 where I'm walking through Kind Heading only and adapt.
